swap_meet/vendor.py

```diff
@@ -1,2 +1,120 @@
 class Vendor:
-    pass
\ No newline at end of file
+# WAVE 1
+    def __init__(self, inventory=None):
+        # if inventory is none then set an empty list by default
+        if inventory is None:
+            inventory = []
+        # attribute (allows us to access inventory later)
+        self.inventory = inventory
+
+
+    # add method
+    def add(self, item):
+        # append the item to the inventory and return item
+        self.inventory.append(item)
+        return item
+    # remove method
+    def remove(self, item):
+        # check if item is not in inventory rhen return False
+        if item not in self.inventory:
+            return False
+        else:
+            # remove the item from the inventory using pop method
+            # (first get index of item) then return the removed item
+            index = self.inventory.index(item)
+            removed_item = self.inventory.pop(index)
+            return removed_item
+
+# WAVE 2
+    def get_by_id(self, id):
+        for item in self.inventory:
+            # check if the item id match with a given id
+            if item.id == id:
+                return item
+        # no matching item in the inventory
+        return None
+
+# WAVE 3
+    # Swapping items method
+    def swap_items(self, other_vendor, my_item, their_item):
+        # if vendors inventory doesn't contain 'my_item' or friends inventory dosent contain 'their_item then return False
+        # also if list empty return false
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        else:
+            # remove my_item from my inventory
+            index = self.inventory.index(my_item)
+            removed_item = self.inventory.pop(index)
+            # add my_item to friends inventory
+            other_vendor.inventory.append(my_item)
+            # remove their_item from friends inventory
+            index = other_vendor.inventory.index(their_item)
+            removed_item = other_vendor.inventory.pop(index)
+            # add their_item to my inventory
+            self.inventory.append(their_item)
```

Comment:
Using `pop` and `append` to swap the items works well, and from a complexity perspective, does no worse than any other approach for an arbitrary swap. However, notice that in the tests provided with the project, there were no assumptions made that the swapped items would end up at the ends of the swapped lists. There are other approaches we could consider if we're open to the swapped items ending up in other locations.

Think about other approaches we could take, and if there are particular situations where this or other approaches could exhibit different performance.